src/components/data-exploration/embedding/Embedding.jsx

```diff
@@ -132,23 +132,33 @@ const Embedding = (props) => {
   }, [focusedExpression]);
 
   useEffect(() => {
-    if (cellSetHierarchy) {
-      const mapping = cellSetHierarchy.reduce((keyToClusterNameMap, rootHierarchy) => {
-        if (rootHierarchy.children.length > 0) {
-          rootHierarchy.children.forEach((child) => {
-            // eslint-disable-next-line no-param-reassign
-            keyToClusterNameMap[child.key] = _.capitalize(rootHierarchy.key);
-          });
-        }
-        return keyToClusterNameMap;
-      }, {});
-      setClusterKeyToNameMap(mapping);
-    }
-
-    if (cellSetProperties) {
-      setCellSetClusters(Object.entries(cellSetProperties).filter(([, cellSet]) => cellSet.type === 'cellSets'));
+    if (selectedCell) {
+      const expressionToDispatch = focusedExpression
+        ? focusedExpression.rawExpression.expression[selectedCell] : undefined;
+
+      // getting the root nodes which are of type cellSets
+      const rootClusterNodes = cellSetHierarchy.map(({ key }) => key)
+        .filter((key) => cellSetProperties[key].type === 'cellSets');
+
+      // getting the cluster properties for every cluster that has the cellId
+      const cellProperties = getContainingCellSetsProperties(Number.parseInt(selectedCell, 10), rootClusterNodes, cellSets);
+
+      const prefixedCellSetNames = [];
+      Object.entries(cellProperties).forEach(([key, clusterProperties]) => {
+        clusterProperties.forEach(({ name, parentNodeKey }) => {
+          prefixedCellSetNames.push(`${cellSetProperties[parentNodeKey].name} : ${name}`);
+        });
+      });
+
+      cellInfoTooltip.current = {
+        cellSets: prefixedCellSetNames,
+        cellName: selectedCell,
```

Comment:
I think it would be good to change all occurrences of `cellName` to `cellId`. Is it possible to do that?